students/dave_arasim/session04/mailroom2.py

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python
+'''Mailroom Part 2 for Session 4, Python 210
+Written by David K. Arasim - 10/28/18'''
+
+#Imported modules
+import platform
+import os
+from collections import defaultdict
+
+#Global variables
+this_os = platform.system()
+quit_option = False
+
+#############################################################################################
+#Function section
+
+def main():
+    global this_os
+    global quit_option
+
+    donor_dict = defaultdict(list)
+    build_donor_dict(donor_dict)    #Build the Donor Dictionary from a text file
+
+    switch_dict = {1: send_thank_you, 2: create_report, 3: send_all_thanks, 4: quit_option_true}
+
+    while not(quit_option):
+        if this_os == 'Windows':
+            os.system('clear')
+        else:
+            os.system('cls')
+
+        print ('*'*10,'  Main Menu  ','*'*10)
+        print()
+        print('1) Send a Thank You to a single donor')
+        print('2) Create a Report')
+        print('3) Send Thank You letters to all donors')
+        print('4) Quit')
+        print()
+        print('Choose an option: ', end='')
+
+        user_choice = input()
+        print()
+
+        try:
+            user_choice = int(user_choice)
+        except ValueError:
+            user_choice = ''
+
+        if user_choice == '':
+            print('Sorry, but that is not a number')
+        else:
+            try:
+                switch_dict.get(user_choice)(donor_dict)
+            except TypeError:
+                print('That option: ', user_choice, ' is out of range', sep='')
```

Comment:
Great to see the same here as well, but it looks to me that the two `try: ...` blocks can be combined into one...? Note that you can have multiple `except ...: ...` clauses for a `try: ...` block...